allennlp/training/trainer.py

```diff
@@ -1,7 +1,7 @@
 import logging
 import os
 import shutil
-from typing import Optional
+from typing import Optional, List  # pylint: disable=unused-import
```

Comment:
Man, every time I see this it makes me want to switch to python 3.6, so we can just put the variable types inline and remove all of these `disable=unused-import` statements.  Not for this PR, though...